### PR TITLE
Allow defining many tags per service

### PR DESCRIPTION
**Pull request by chalasr**
For e.g. a `logger.foo_channel` +`listener.stream_closed` tagged service

### CODE REVIEW

File: src/Container.js [hunk 1]
```diff
@@ -107,7 +108,7 @@ class Container {
   getTaggedService(tag) {
     return Array
       .from(this.services.values())
-      .filter(definition => definition.tag === tag)
+      .filter(definition => tag in definition.tags)
```
Comment (Tom32i) [inline src/Container.js]: The `in` notation only work for object, you need `includes` here:
`definition.tags.includes(tag)`

Comment (Tom32i) [inline src/Container.js]: An thank you for your interest 🤗

Comment (chalasr) [inline src/Container.js]: Fixed!

File: src/Container.js [hunk 2]
```diff
@@ -23,11 +23,11 @@ class Container {
    * @param {String} name
    * @param {Scalar|Function} value
    * @param {Array} dependencies
-   * @param {String} tag
+   * @param {String|Array} tags
```
Comment (ogizanagi) [inline src/Container.js]:
```diff
- @param {String|Array} tags
+ @param {String|String[]} tags
```

?

File: src/Container.js [hunk 3]
```diff
@@ -39,11 +39,12 @@ class Container {
    * @param {String} name
    * @param {Function} classname
    * @param {Array} dependencies
-   * @param {String} tag
+   * @param {String|Array} tags
```
Comment (ogizanagi) [inline src/Container.js]: (Same)

File: src/Container.js [hunk 4]
```diff
@@ -39,11 +39,12 @@ class Container {
...
    */
-  registerDefinition(name, classname, dependencies = [], tag = null) {
+  registerDefinition(name, classname, dependencies = [], tags = []) {
+    tags = typeof (tags) === 'string' ? [tags] : tags;
```
Comment (ogizanagi) [inline src/Container.js]:
```diff
- tags = typeof (tags) === 'string' ? [tags] : tags;
+ tags = typeof tags === 'string' ? [tags] : tags;
```

(unecessary parenthesis for an operator) ?

File: src/Container.js (hunk 2 above)
Comment (chalasr) [inline src/Container.js]: 👍  (I'm in the wild here)

File: src/Container.js (hunk 4 above)
Comment (chalasr) [inline src/Container.js]: agreed, but it's consistent with https://github.com/Elao/container.js/blob/master/src/Container.js#L139

**Comment by chalasr** [comment]
Tests added

**Comment by chalasr** [comment]
Bump

**Pull request merged**